company/views.py

```diff
@@ -147,6 +152,31 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class CompanyMoveProducts(LoginPermissionRequiredMixin, MessageMixin, FormView, DetailView):
+    permission_required = 'company.change_company'
+    template_name = 'company/company_move_products.html'
+    form_class = MoveProductsForm
+    queryset = Company.objects.all()
+
+    def form_valid(self, form):
+        pk = self.kwargs['pk']
+        success, failed = form.save(pk)
```

Comment:
Przyjeło się, że form.save() zwraca ID zedytowanego obiektu lub nowoutworzonego.

Reply:
Jeśli formularz potrzebuje parametrów, to należy rozszerzyć get_form_kwargs i dodać wymagane obiekty do konstruktora.